packages/yourloops/pages/profile/profile.tsx

```diff
@@ -342,162 +280,66 @@ const ProfilePage = (props: ProfilePageProps): JSX.Element => {
-            <strong className={classes.uppercase}>{t("preferences")}</strong>
-          </Box>
-
-          <Box className={classes.inputContainer}>
-            <FormControl className={`${classes.formControl} ${classes.halfWide}`}>
-              <InputLabel id="profile-units-input-label">{t("units")}</InputLabel>
-              <Select
-                disabled={role === UserRoles.patient}
-                labelId="unit-selector"
-                id="profile-units-selector"
-                value={unit}
-                onChange={createHandleSelectChange(setUnit)}
-              >
-                <MenuItem id="profile-units-mmoll" value={Units.mole}>
-                  {Units.mole}
-                </MenuItem>
-                <MenuItem id="profile-units-mgdl" value={Units.gram}>
-                  {Units.gram}
-                </MenuItem>
-              </Select>
-            </FormControl>
-            <FormControl className={`${classes.formControl} ${classes.halfWide}`}>
-              <InputLabel id="profile-language-input-label">{t("language")}</InputLabel>
-              <Select
-                labelId="locale-selector"
-                id="profile-locale-selector"
-                value={lang}
-                onChange={createHandleSelectChange(setLang)}>
-                {availableLanguageCodes.map((languageCode) => (
-                  <MenuItem id={`profile-locale-item-${languageCode}`} key={languageCode} value={languageCode}>
-                    {getLangName(languageCode)}
-                  </MenuItem>
-                ))}
-              </Select>
-            </FormControl>
-          </Box>
-
-          {showFeedback &&
-            <ConsentFeedback
-              id="profile"
-              userRole={role}
-              checked={feedbackAccepted}
-              style={{ marginLeft: -9, marginRight: 0, marginTop: "1em", marginBottom: 0 }}
-              onChange={() => setFeedbackAccepted(!feedbackAccepted)}
-            />
-          }
+          <PreferencesForm
+            classes={classes}
+            feedbackAccepted={feedbackAccepted}
+            lang={lang}
+            role={role}
+            unit={unit}
+            setFeedbackAccepted={setFeedbackAccepted}
+            setLang={setLang}
+            setUnit={setUnit}
+          />
 
           <Box display="flex" justifyContent="flex-end" my={3}>
             <Button
               id="profile-button-cancel"
-              onClick={onCancel}
+              onClick={() => history.push(props.defaultURL)}
             >
               {t("button-cancel")}
             </Button>
-            <Button
-              id="profile-button-save"
-              variant="contained"
-              disabled={!canSave}
-              color="primary"
-              onClick={onSave}
-              className={classes.button}
-            >
-              {t("button-save")}
-            </Button>
+            <ProgressIconButtonWrapper inProgress={saving}>
```

Comment:
This is new, right?

Reply:
yeah, a progress loader. And also disable the button when making request, currently you can still click on it when saving